sklearn/linear_model/_base.py

```diff
@@ -191,6 +195,26 @@ def _rescale_data(X, y, sample_weight):
                                   shape=(n_samples, n_samples))
     X = safe_sparse_dot(sw_matrix, X)
     y = safe_sparse_dot(sw_matrix, y)
+
+    if sparse_X:
+        if order == 'F':
+            X = sparse.csc_matrix(X)
+        else:
+            X = sparse.csr_matrix(X)
```

Comment:
Maybe,
```py
sparse_format = "csc" if order == "F" else "csr"

X = X.asformat(sparse_format, copy=False)
```

that should avoid copies if the sparse matrix is already in the right format..